Ensure persist_block persists tx and receipts (#1932)
ethereum/py-evm: newsfragments/1925.feature.rst

Add a new ``persist_unexecuted_block`` API to ``ChainDB``. This API should be used to persist

> blocks that the EVM did not execute before it stores them. The API is needed to allow to build

Reviewer: ```suggestion
a block without executing the EVM on it. The API is used by
```
